packages/adminrouter/extra/src/includes/http/common.conf

```diff
@@ -1,9 +1,16 @@
-# Enable VTS module for metrics
-# Set the shared memory zone size to 32MB to be sure to
-# contain all metrics in the VTS module configuration.
-# If the VTS module configuration enables dynamic metric
-# keys this needs to be increased to accommodate for that.
-vhost_traffic_status_zone shared:vhost_traffic_status:32m;
+# Enable VTS module for metrics.
+# The shared memory zone size must be sufficiently large to ensure that all
+# metrics that could potentially be generated by the VTS module's
+# vhost_traffic_status_filter_by_set_key directive can be stored. If the
+# shared memory zone limit is hit Admin Router will crash. The metrics data
```

Comment:
Do we know what this crash would look like? Are we likely to get an error message, or will the logs just show Admin Router restarting? Is there anything we can do to make it obvious if this failure occurs? Can we log how close we are to the limit?

Reply:
We have a metric in place that is displayed in the Grafana dashboards for the used shared memory size. Ideally we set up an alarm for this later on. I have not actually tried to crash Admin Router, but due to how DC/OS handles Admin Router crashes I suspect that Admin Router is restarted by the systemd unit and the in-memory metric state is lost.
If you want I can try it out I suppose.

Reply:
Thanks for suggesting this @jongiddy. I ran a couple of tests and it appears that the module does not crash Admin Router. When reaching about 90% of the shared memory max size the VTS module stops to create new metrics nodes for new unique URIs, however the requests themselves are still counted. This means we definitely need to incorporate a LRU mechanism but it also means that we don't need to worry about the crash case. I think we should still set up an alert later at maybe around 80%.

Reply:
This is no longer the case, we employ an LRU mechanism now.